src/main/java/org/jabref/model/entry/BibEntryTypeBuilder.java

```diff
@@ -20,17 +23,25 @@
 public class BibEntryTypeBuilder {
 
     private EntryType type = StandardEntryType.Misc;
-    private SequencedSet<BibField> fields = new LinkedHashSet<>();
+
     private SequencedSet<OrFields> requiredFields = new LinkedHashSet<>();
+    private SequencedSet<BibField> optionalFields = new LinkedHashSet<>();
+
+    private Set<Field> seenFields = new HashSet<>();
 
     public BibEntryTypeBuilder withType(EntryType type) {
         this.type = type;
         return this;
     }
 
     public BibEntryTypeBuilder withImportantFields(SequencedSet<Field> newFields) {
-        this.fields = Streams.concat(fields.stream(), newFields.stream().map(field -> new BibField(field, FieldPriority.IMPORTANT)))
-                             .collect(Collectors.toCollection(LinkedHashSet::new));
+        List<Field> containedFields = containedInSeenFields(newFields);
+        if (!containedFields.isEmpty()) {
```

Comment:
Isn't the idea of a set to prevent duplicates. Why the check here

Reply:
Because of "throw new IllegalArgumentException". To warn dumb callers.